src/drivers/dmic.c

```diff
@@ -1150,6 +1150,13 @@ static int dmic_set_config(struct dai *dai, struct sof_ipc_dai_config *config)
 		goto finish;
 	}
 
+	if (!(prm->fifo_bits_a == 16 || prm->fifo_bits_a == 32) ||
```

Comment:
easier to read two conditional statements like if (a != 16 && a != 32) { error}, where we could also have different error codes for each DMIC.